packages/react/src/__tests__/ReactProfiler-test.internal.js

```diff
@@ -3016,6 +3098,8 @@ describe('Profiler', () => {
         expect(Scheduler).toHaveYielded(['Component:true']);
         expect(onNestedUpdateScheduled).not.toHaveBeenCalled();
       });
+
+      // TODO Add hydration tests to ensure we don't have false positives called.
```

Comment:
There might be other edge cases related to hydration or Suspense that I haven't thought about yet. This note is for me to follow up and think about them more.